lib/catalog/martyrology.ts

```diff
@@ -3321,7 +3323,10 @@ export const Martyrology: { catalog: MartyrologyCatalog } = {
       titles: [Title.Religious],
     },
     passion_of_saint_john_the_baptist: {
+      // Note: unlike the celebration of the Nativity of Saint John the Baptist, here his title of Martyr is indeed indicated.
+      // Therefore, this celebration will be celebrated in red (src: GIRM, 308b).
       name: 'Passion of Saint John the Baptist',
+      titles: [Title.Martyr],
```

Comment:
I believe this is not the best solution.

While St John the Baptist was a martyr, the name of this celebration does not include the `Martyr` title, [1] therefore IMHO we should not return `Title.Martyr` for this celebration. At least not until we (1) assign/choose all the titles in each celebration definition in calendars; and (2) we generate the celebration names from separated l10n data.

IMHO defining `color` explicitly is a better solution in this case. Or we could check for the celebration ID and implicitly define the color for this celebration.

[1] The celebration is, as I call it, one of an _event_ (the passion/martyrdom of St John the Baptist) and not a personal celebration.

Reply:
I didn't have the same reasoning, but your point of view is interesting.

On my side, I tried to strictly follow the rules of the Missal to be as objective as possible: if a saint is a martyr, they are celebrated in red; if not, it is in white. Therefore, if a memorial is celebrated in red, it means that the person is indeed a martyr (unless there is an explicit exception described in the Missal).

Indeed, the word "Martyr" is not directly in the title of this memorial, but in this context, the term "Passion" (used in Latin and English) remains a synonym for "Martyr". FYI, the word "Martyr" is explicitly used in French. So, I relied on the traditional practices of the Church—which celebrates this memorial in red—to corroborate adding the title "Martyr".

So, I would say, regardless of which rule we want to follow here: based on the terminology of the memorial's title, or based on the analysis of the Missal's rules (although I prefer the latter, which I find more objective), as long as this liturgical day is celebrated in red, it works for me. I'll leave the final word to you on this 😉

Reply:
> On my side, I tried to strictly follow the rules of the Missal to be as objective as possible: if a saint is a martyr, they are celebrated in red; if not, it is in white. Therefore, if a memorial is celebrated in red, it means that the person is indeed a martyr (unless there is an explicit exception described in the Missal).

I have no objection to this reasonsing of yours. All I disagree is to add a title to a celebration which does not use it in its celebration name.

While your solution does not have any downside in the current romcal implementation, however, once we generate the celebration name from its metadata (see #154), it will be an issue. That’s I think your solution (adding a `Title.Martyr` to this celebration) is not the best solution.

In the end, IMHO we should add a custom rule to the romcal core (or however you call it; the place where we do the calendar generation itself), as this celebration is inscribed in the GRC, thus celebration everywhere. That said, an easier solution is to use `color` explicitly to override this.

> Indeed, the word "Martyr" is not directly in the title of this memorial, but in this context, the term "Passion" (used in Latin and English) remains a synonym for "Martyr". FYI, the word "Martyr" is explicitly used in French. So, I relied on the traditional practices of the Church—which celebrates this memorial in red—to corroborate adding the title "Martyr".

Yeah, I agree with `martyr` and `passion` in this sense are related terms, however, they are not synonyms. In the end, it does matter whether they are synonyms on not, the the _usage_ is what matters here IMHO:

- `passion` is used in the celebration name;
- `martyr` (and thus `Title.Martyr`) is used in the celebration metadata.

The latter is incorrect, as the celebration is not a _personal_ celebration. We assign titles to persons (saints, blesseds), not to a non-personal celebrations (i.s. events).

I agree that St John the Baptist _is_ a martyr. Therefore, we should add `Title.Martyr` to `john_the_baptist_martyr` in `martyrology.ts` (or the future file with all the celebration metadata), however, IMHO it should not be used for this celebration.

Reply:
I'm totally on board with you on the terminology and localization front. So, let's go ahead and use the `color` property here. I'll open a PR to get that sorted.

Reply:
The fix is here: https://github.com/romcal/romcal/pull/584